### PR TITLE
CB-7105 Add resource based Auth for Redbeams api

### PR DESCRIPTION
**Pull request by horadla23**

### CODE REVIEW

**Comment by horadla23** [comment]
tested with (mowdev ums):
- cli redbeams describe with owner and non-owner
- cli redbeams delete with owner and non-owner
- cli redbeams list
- sdx create with database
- sdx delete
- distrox create with database
- distrox stop with database
- distrox start with database
- distrox delete

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/dbconfig/DatabaseConfigService.java [hunk 1]
```diff
@@ -271,4 +280,25 @@ public Long getResourceIdByResourceCrn(String resourceCrn) {
...
     }
+
+    @Override
+    public AuthorizationResourceType getResourceType() {
+        return AuthorizationResourceType.DATABASE;
+    }
+
+    @Override
+    public String getResourceCrnByResourceName(String resourceName) {
+        return getByName(resourceName).getResourceCrn().toString();
```
Comment (foldik) [inline redbeams/src/main/java/com/sequenceiq/redbeams/service/dbconfig/DatabaseConfigService.java:291]: Can you get just the crn from the database?

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/dbconfig/DatabaseConfigService.java [hunk 2]
```diff
@@ -271,4 +280,25 @@ public Long getResourceIdByResourceCrn(String resourceCrn) {
...
+    }
+
+    @Override
+    public List<String> getResourceCrnListByResourceNameList(List<String> resourceNames) {
+        return repository.findByNameIn(resourceNames).stream()
```
Comment (foldik) [inline redbeams/src/main/java/com/sequenceiq/redbeams/service/dbconfig/DatabaseConfigService.java:296]: Can you get just the crns from the database?

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java [hunk 3]
```diff
@@ -320,6 +325,22 @@ boolean validateDatabaseName(String databaseName) {
 
     @Override
     public AuthorizationResourceType getResourceType() {
-        return AuthorizationResourceType.DATALAKE;
+        return AuthorizationResourceType.DATABASE_SERVER;
+    }
+
+    @Override
+    public String getResourceCrnByResourceName(String resourceName) {
+        return repository.findByName(resourceName).get().getResourceCrn().toString();
```
Comment (foldik) [inline redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java:333]: Can you get just the crn from the database?

File: redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java [hunk 4]
```diff
@@ -320,6 +325,22 @@ boolean validateDatabaseName(String databaseName) {
...
+    }
+
+    @Override
+    public List<String> getResourceCrnListByResourceNameList(List<String> resourceNames) {
+        return repository.findByNameIn(resourceNames).stream()
```
Comment (foldik) [inline redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java:338]: Can you get just the crn from the database?

Comment (horadla23) [reply redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java]: I made repository methods for these as suggested

**Pull request merged**